### DISCUSSION TITLE
Should landmark registration throw error if landmarks is an empty list?

### DISCUSSION THREAD

**Discussion by hovinh** [GAA-UAM/scikit-fda#394] [Q&A]
Hi team,

I have used landmark registration for my trace alignment and noticed a strange behaviour. If I accidentally assign the argument `landmarks = [[]]` in `registration.landmark_registration_warping`, it still processed normally. The trace I provided is an anomaly trace, hence the defined landmarks were unable to extract. Based on the output trace and warping plot (a straight diagonal line), I believe it simply do no warping at all. 

Do you think this is intended? Personally, I would like to have a checking if `landmarks` have the same number of elements with `location` argument  and should contain at least 1 element, or a warning for users. And could you please share how this alignment method behaves when no landmarks provided, would it take the first and last element as landmarks?

Thank you.

Sincerely,
Vinh

**Comment by vnmabus** [comment]
Sorry but in the develop version there is a check for that (and the function is called `landmark_elastic_registration_warping`). Also you didn't provide any trace.

May you check if that works in develop, and show an example that fails if that does not work?

**Reply by hovinh** [reply]
Hi, I checked the develop branch:
- It's true that if `landmarks` is empty and `location` is non-empty, there is error raised.
```
Error: ValueError: Number of landmark locations should be equal than the number of landmarks (2) != (0)
```
- If both `landmarks` and `location` are empty, then no error thrown. This is actually my case. Put it simply, `location` are determined as median from  `landmarks`, hence if one is empty, the other follows suite. I designed it that way to do something similar to a groupwise landmark registration.
Example:

```python
import matplotlib.pyplot as plt
import numpy as np

import skfda
fd = skfda.datasets.make_multimodal_samples(
    n_samples=4,
    n_modes=2,
    std=0.002,
    mode_std=0.005,
    random_state=1,
)
fd.plot()
plt.show()
```
![image](https://user-images.githubusercontent.com/7614509/142580186-4e4fa3a4-4b20-4497-bce2-e373c9020729.png)


```python
landmarks=[[], [], [], []]
warping = skfda.preprocessing.registration.landmark_elastic_registration_warping(
    fd,
    landmarks,
    location=[],
)

# Plots warping
fig = warping.plot()
```

![image](https://user-images.githubusercontent.com/7614509/142580349-a9494428-dd4f-43a4-ad96-e4b064370a23.png)
